presto-main/src/test/java/io/prestosql/operator/TestTopNRowNumberOperator.java

```diff
@@ -93,30 +94,30 @@ public void tearDown()
     @Test(dataProvider = "hashEnabledValues")
     public void testPartitioned(boolean hashEnabled)
     {
-        RowPagesBuilder rowPagesBuilder = rowPagesBuilder(hashEnabled, Ints.asList(0), BIGINT, DOUBLE);
+        RowPagesBuilder rowPagesBuilder = rowPagesBuilder(hashEnabled, Ints.asList(0), VARCHAR, DOUBLE);
         List<Page> input = rowPagesBuilder
-                .row(1L, 0.3)
-                .row(2L, 0.2)
-                .row(3L, 0.1)
-                .row(3L, 0.91)
+                .row("a", 0.3)
+                .row("b", 0.2)
+                .row("c", 0.1)
+                .row("c", 0.91)
                 .pageBreak()
-                .row(1L, 0.4)
+                .row("a", 0.4)
                 .pageBreak()
-                .row(1L, 0.5)
-                .row(1L, 0.6)
-                .row(2L, 0.7)
-                .row(2L, 0.8)
+                .row("a", 0.5)
+                .row("a", 0.6)
+                .row("b", 0.7)
+                .row("b", 0.8)
                 .pageBreak()
-                .row(2L, 0.9)
+                .row("b", 0.9)
                 .build();
 
         TopNRowNumberOperatorFactory operatorFactory = new TopNRowNumberOperatorFactory(
                 0,
                 new PlanNodeId("test"),
-                ImmutableList.of(BIGINT, DOUBLE),
+                ImmutableList.of(VARCHAR, DOUBLE),
```

Comment:
Why would that fail with previous `TopNRowNumberOperatorCode` code? Was it because `BIGINT` and `DOUBLE` comparisons were compatible?

Reply:
This did not fail before, which was the problem -- when it should have. BIGINT and DOUBLE comparisons are entirely the same, except when it comes to the special Double values like NaN etc. It was previously using BIGINT to process DOUBLE data. VARCHAR makes this much more apparent.